File: blog-java/blog-web/src/main/java/com/blog/controller/UserController.java
package com.blog.controller;

import com.blog.common.Result;
import com.blog.dto.LoginDTO;
import com.blog.dto.RegisterDTO;
import com.blog.service.UserService;
import com.blog.vo.LoginVO;
import com.blog.vo.UserInfoVO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@Tag(name = "用户管理")
@RestController
public class UserController {

    @Autowired
    private UserService userService;

    @Operation(summary = "用户登录")
    @PostMapping("/login")
    public Result<LoginVO> login(@Valid @RequestBody LoginDTO loginDTO) {
        return Result.ok(userService.login(loginDTO));
    }

    @Operation(summary = "用户登出")
    @PostMapping("/logout")
    public Result<Void> logout() {
        userService.logout();
        return Result.ok();
    }

    @Operation(summary = "用户注册")
    @PostMapping("/register")
    public Result<Void> register(@Valid @RequestBody RegisterDTO registerDTO) {
        userService.register(registerDTO);
        return Result.ok();
    }

    @Operation(summary = "获取当前用户信息")
    @GetMapping("/user/info")
    public Result<UserInfoVO> getUserInfo() {
        return Result.ok(userService.getUserInfo());
    }

    @Operation(summary = "更新用户信息")
    @PutMapping("/user/info")
    public Result<Void> updateUserInfo(@RequestBody UserInfoVO userInfoVO) {
        userService.updateUserInfo(userInfoVO);
        return Result.ok();
    }

    @Operation(summary = "获取管理员信息")
    @GetMapping("/user/admin")
    public Result<UserInfoVO> getAdminInfo() {
        return Result.ok(userService.getAdminInfo());
    }
}

File: blog-go/controller/user_controller.go
package controller

import (
	"blog-go/middleware"
	"blog-go/model"
	"blog-go/response"
	"blog-go/service"

	"github.com/gin-gonic/gin"
)

type UserController struct {
	userService *service.UserService
}

func NewUserController() *UserController {
	return &UserController{
		userService: &service.UserService{},
	}
}

func (c *UserController) Login(ctx *gin.Context) {
	var dto model.LoginDTO
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		response.BadRequest(ctx, "参数错误")
		return
	}

	result, err := c.userService.Login(&dto)
	if err != nil {
		response.Error(ctx, 500, err.Error())
		return
	}

	response.Success(ctx, result)
}

func (c *UserController) Logout(ctx *gin.Context) {
	userID := middleware.GetUserID(ctx)
	if err := c.userService.Logout(userID); err != nil {
		response.Error(ctx, 500, "登出失败")
		return
	}
	response.Success(ctx, nil)
}

func (c *UserController) Register(ctx *gin.Context) {
	var dto model.RegisterDTO
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		response.BadRequest(ctx, "参数错误")
		return
	}

	if err := c.userService.Register(&dto); err != nil {
		response.Error(ctx, 500, err.Error())
		return
	}

	response.Success(ctx, nil)
}

func (c *UserController) GetUserInfo(ctx *gin.Context) {
	userID := middleware.GetUserID(ctx)
	userInfo, err := c.userService.GetUserInfo(userID)
	if err != nil {
		response.Error(ctx, 500, "获取用户信息失败")
		return
	}
	response.Success(ctx, userInfo)
}

func (c *UserController) UpdateUserInfo(ctx *gin.Context) {
	userID := middleware.GetUserID(ctx)
	var dto model.UserInfoDTO
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		response.BadRequest(ctx, "参数错误")
		return
	}

	if err := c.userService.UpdateUserInfo(userID, &dto); err != nil {
		response.Error(ctx, 500, "更新用户信息失败")
		return
	}

	response.Success(ctx, nil)
}

func (c *UserController) GetAdminInfo(ctx *gin.Context) {
	userInfo, err := c.userService.GetAdminInfo()
	if err != nil {
		response.Error(ctx, 500, "获取管理员信息失败")
		return
	}
	response.Success(ctx, userInfo)
}
